docs/content/docs/next-js.md

```diff
@@ -0,0 +1,142 @@
+packages:
+    - apps
+```
+
+## AdonisJS
+
+Now let's do a `cd` to our `apps` folder and create a adonisjs project:
+
+```sh
+pnpm create adonisjs@latest backend
+```
+
+For more information on creating an adonisjs project, check out [adonisjs](https://docs.adonisjs.com/guides/getting-started/installation)
+
+```
+acme
+├── package.json
+├── apps
+└── apps/backend
+```
+
+## Tuyau
+
+For installation of the Tuyau framework, check out the [installation](/docs/installation) guide.
+
+## Next.js
+
+To create a Next.js project, run the following command in `./acme/apps` folder:
+
+```sh
+pnpx create-next-app@latest frontend --typescript
+```
+
+For more information on creating a Next.js project, check out [create-next-app](https://nextjs.org/docs/getting-started/installation)
+
+```
+acme
+├── package.json
+├── apps
+├── apps/backend
+└── apps/frontend
+```
+
+### Importing the API Definition
+
+To import the API definition from your AdonisJS project generated by the `tuyau:generate` command, you must import the `.adonisjs/index.ts` file in your Next.js project.
+
+For that please follow the [installation](/docs/installation) guide from [here](/docs/installation#sharing-the-api-definition).
+
+### Configuration
+
+As we are directly importing `.ts` files into our Next.js project, we must configure the `tsconfig.json` and `next.config.ts` files.
+
+```ts
+// next.config.ts
+import type { NextConfig } from 'next';
+
+const nextConfig: NextConfig = {
+  reactStrictMode: true,
+  transpilePackages: ['@acme/backend'],
+  webpack: (config) => {
+    config.resolve.extensionAlias = {
+      '.js': ['.ts', '.tsx', '.js', '.jsx'],
+      '.mjs': ['.mts', '.mjs'],
+      '.cjs': ['.cts', '.cjs'],
+    };
+    return config;
+  },
+};
+
+export default nextConfig;
+```
+
+Decorators are not enabled by default in TypeScript, So we need to enable them in `tsconfig.json` else we'll see errors during `tsc --noEmit` (typecheck).
+
+```json
+// tsconfig.json
+{
+  "compilerOptions": {
+    "experimentalDecorators": true
+  }
+}
+```
+
+---
+
+We have successfully created a monorepo with AdonisJS, Next.js and Tuyau, but let's check if everything is working as expected.
+
+Add the `typecheck` script to the `package.json` of `./acme/apps/frontend`:
+
+```json
+// package.json
+{
+  "scripts": {
+    "typecheck": "tsc --noEmit"
+  }
+}
+```
+
+Run the `typecheck` script and see if everything is working as expected.
+
+If you see any errors coming from `../backend` folder, check them and if they are related to missing types, reference them in the `./.adonisjs/index.ts` file like this:
+
+```ts
+/// <reference path="../adonisrc.ts" />
+/// <reference path="../config/auth.ts" />
+
+export * from './api.js'
+```
+
+As your project grows you will need to add more references to the types in the `./.adonisjs/index.ts` file.
+
+---
+
+You can find an example of AdonisJS, Next.js and Tuyau in the [examples](https://github.com/Julien-R44/tuyau/tree/main/examples/nextjs) folder.
```

Comment:
We can probably leave it here too, but I think it should also be added to the top of the page. It's probably what people are looking for first

Reply:
Yeap, you are right.